planner/core/testdata/plan_suite_in.json

```diff
@@ -596,6 +596,19 @@
       "with cte1 as (with cte2 as (with cte3 as (select  /*+ MERGE() */  * from t2) select  /*+ MERGE() */  * from cte3) select * from cte2,(select * from t1) ttt) select * from cte1,(select * from t3) ttw;"
     ]
   },
+  {
+    "name": "TestInlineCTE",
+    "cases": [
+      "set tidb_opt_inline_cte=1;", // enable inline CTE
+      "with cte as (select * from t) select * from cte;", // inline
```

Comment:
Can we add test cases where :
1. CTE has multiple consumers,
2. 2 CTEs, CTE_1 is consumer of CTE_2, and there is a join chain: CTE_1 join CTE_1 join CTE_2

Reply:
done